### ISSUE TITLE
Endless loop while parsing invalid eml data.

### ISSUE THREAD

**Issue by rboen** [jstedfast/MimeKit#384]
Hi there,

caused by foolish bug I did not pass the valid EML-Data to MimeMessage.Load but the string "System.Streams.MemoryStream" itsself. As a consequence the parser routine of MimeMessage got stuck in an infinite loop. It would be nice, if this kind of maleformed data will be rejected by throwing an exception.

Code to reproduce:

```
            var emlData = "System.Streams.MemoryStream";
            var stream = new MemoryStream();
            var writer = new StreamWriter(stream);
            writer.Write(emlData);
            writer.Flush();
            stream.Position = 0;
            var mimeMessage = MimeMessage.Load(stream);
```

Thank you very much for your incredible work on this great open source project!!!

Best regards,

Ralf

**Comment by jstedfast** [comment]
Seems to work fine with the latest revision of the code. Make sure you're using the latest release.

**Comment by springy76** [comment]
So please release NuGet packages. I just tested the latest **release* 2.0.1 of MimeKitLite and it just got stuck.
(Just wanted to report this error which occured for the input "{}" to me.)

**Comment by jstedfast** [comment]
Can you confirm that it's fixed in the latest revision on github? I was pretty sure the fix for this was already in 2.0.1.

**Comment by springy76** [comment]
Using the current master branch (last commit 2018-02-28) both my and rboen's code `throw new FormatException ("Failed to parse message headers.");` immediately.

**Comment by jstedfast** [comment]
Ah, looks like the fix was commit 801ac4777902a6d104d9e1f76cf35f096d0b3a5a. I thought it was an earlier commit.

I'll make a release this weekend if I don't get a chance sooner.
